File: common/src/main/java/com/tnt_man_inc/jarm_3/blocks/RubyBlock.java
package com.tnt_man_inc.jarm_3.blocks;

import me.shedaniel.architectury.registry.RegistrySupplier;
import net.minecraft.world.item.Item;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockBehaviour;
import net.minecraft.world.level.material.Material;

import static com.tnt_man_inc.jarm_3.Registries.BLOCKS;
import static com.tnt_man_inc.jarm_3.Registries.ITEMS;


public class RubyBlock {
    public static final RegistrySupplier<Block> RUBY_BLOCK = BLOCKS.register("ruby_block", () -> new Block(BlockBehaviour.Properties.of(Material.HEAVY_METAL)));
    public static final RegistrySupplier<Item> RUBY_BLOCK_ITEM = ITEMS.register("ruby_block", () -> new BaseBlockItem(RUBY_BLOCK.get(), new Item.Properties()));
    public static void init() {}
}

File: common/src/main/kotlin/com/tnt_man_inc/jarm_3/blocks/RubyBlock.kt
package com.tnt_man_inc.jarm_3.blocks

import com.tnt_man_inc.jarm_3.Registries
import com.tnt_man_inc.jarm_3.items.NISGDApple
import net.minecraft.world.item.Item
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.state.BlockBehaviour
import net.minecraft.world.level.material.Material
import java.lang.AssertionError

object RubyBlock {
    val RUBY_BLOCK = Registries.BLOCKS.register("ruby_block") { Block(BlockBehaviour.Properties.of(Material.HEAVY_METAL)) }
    val RUBY_BLOCK_ITEM = Registries.ITEMS.register("ruby_block") { BaseBlockItem(RUBY_BLOCK.get(), Item.Properties()) }
    fun init() {}
}